src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java

```diff
@@ -0,0 +1,122 @@
+                SCANNER_SNYK_ENABLED.getPropertyName(),
+                "true",
+                IConfigProperty.PropertyType.BOOLEAN,
+                "snyk");
+        qm.createConfigProperty(SCANNER_ANALYSIS_CACHE_VALIDITY_PERIOD.getGroupName(),
+                SCANNER_ANALYSIS_CACHE_VALIDITY_PERIOD.getPropertyName(),
+                "86400",
+                IConfigProperty.PropertyType.STRING,
+                "cache");
+        qm.createConfigProperty(SCANNER_SNYK_API_TOKEN.getGroupName(),
+                SCANNER_SNYK_API_TOKEN.getPropertyName(),
+                DataEncryption.encryptAsString("token"),
+                IConfigProperty.PropertyType.STRING,
+                "token");
+        qm.createConfigProperty(SCANNER_SNYK_ORG_ID.getGroupName(),
+                SCANNER_SNYK_ORG_ID.getPropertyName(),
+                "orgid",
+                IConfigProperty.PropertyType.STRING,
+                "orgid");
+        qm.createConfigProperty(SCANNER_SNYK_API_VERSION.getGroupName(),
+                SCANNER_SNYK_API_VERSION.getPropertyName(),
+                "version",
+                IConfigProperty.PropertyType.STRING,
+                "version");
+        qm.createConfigProperty(SCANNER_SNYK_BASE_URL.getGroupName(),
+                SCANNER_SNYK_BASE_URL.getPropertyName(),
+                "http://localhost:1080",
+                IConfigProperty.PropertyType.STRING,
+                "url");
+        Component component = new Component();
+        component.setPurl(new PackageURL("pkg:maven/com.fasterxml.woodstox/woodstox-core@5.0.0"));
+        component.setPurlCoordinates("pkg:maven/com.fasterxml.woodstox/woodstox-core@5.0.0");
+        List<Component> list = new ArrayList<>();
+        list.add(component);
+        SnykAnalysisTask task = new SnykAnalysisTask();
+        SnykAnalysisEvent event = new SnykAnalysisEvent(component);
+        task.inform(event);
+        //task.apiBaseUrl = "http://localhost:1080";
+        //task.analyze(list);
+
+
+        final Consumer<Vulnerability> assertVulnerability = (vulnerability) -> {
+            Assert.assertNotNull(vulnerability);
+            Assert.assertFalse(StringUtils.isEmpty(vulnerability.getTitle()));
+            Assert.assertFalse(StringUtils.isEmpty(vulnerability.getDescription()));
+            Assert.assertEquals("CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:L", vulnerability.getCvssV3Vector());
+            Assert.assertEquals(Severity.CRITICAL, vulnerability.getSeverity());
+            Assert.assertNotNull(vulnerability.getCreated());
+            Assert.assertNotNull(vulnerability.getAliases());
+            Assert.assertEquals(1, vulnerability.getAliases().size());
+            Assert.assertEquals("A", vulnerability.getAliases().get(0).getCveId());
+        };
+
+        Vulnerability vulnerability = qm.getVulnerabilityByVulnId("SNYK", "SNYK-JAVA-COMFASTERXMLWOODSTOX-2928754", true);
+        assertVulnerability.accept(vulnerability);
+
```

Comment:
There should be an assertion that the component `component` has been marked as vulnerable to `vulnerability`. You can use [`qm.contains(Vulnerability, Component)`](https://github.com/DependencyTrack/dependency-track/blob/master/src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java#L261-L278) for this.

Reply:
Done